reana_workflow_controller/rest/utils.py

```diff
@@ -234,14 +234,18 @@ def delete_workflow(workflow, all_runs=False, workspace=False):
                     .all()
                 )
             for workflow in to_be_deleted:
+                # 1. Stop open interactive sessions
                 int_session = workflow.sessions.first()
                 if int_session:
                     kwrm = KubernetesWorkflowRunManager(workflow)
                     kwrm.stop_interactive_session(int_session.id_)
 
                 if workspace:
+                    # 2. disable retention rules, as the workspace will be deleted
+                    workflow.inactivate_workspace_retention_rules()
+                    # 3. delete the workspace
```

Comment:
Minor comment: wouldn't it be better to change the order of 2 and 3 so that the actual deletion of files will occur first? Thinking about a possible corner case what if something goes wrong with the retention rule inactivation: at least the files would be deleted by the time we get an exception from that function if we switch the order. (Assuming that the most frequent use case is for the researcher to liberate some disk space.)